File: io-hotmoka-network/src/main/java/io/hotmoka/network/internal/models/requests/JarStoreInitialTransactionRequestModel.java
package io.hotmoka.network.internal.models.requests;

import java.util.List;

import io.hotmoka.beans.requests.JarStoreInitialTransactionRequest;
import io.hotmoka.network.exception.GenericException;
import io.hotmoka.network.internal.models.storage.TransactionReferenceModel;
import io.hotmoka.network.internal.util.StorageResolver;

public class JarStoreInitialTransactionRequestModel extends InitialTransactionRequestModel {
    private String jar;
    private List<TransactionReferenceModel> dependencies;

    public void setJar(String jar) {
        this.jar = jar;
    }

    public void setDependencies(List<TransactionReferenceModel> dependencies) {
        this.dependencies = dependencies;
    }

    public JarStoreInitialTransactionRequest toBean() {
    	 if (jar == null)
    		 throw new GenericException("Transaction rejected: Jar missing");

         return new JarStoreInitialTransactionRequest(decodeBase64(jar), StorageResolver.resolveJarDependencies(dependencies));
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/requests/JarStoreInitialTransactionRequestModel.kt
package io.hotmoka.network.thin.client.models.requests

import io.hotmoka.network.thin.client.models.values.TransactionReferenceModel

/**
 * The model of an initial jar store transaction request.
 */
class JarStoreInitialTransactionRequestModel(
    val jar: String,
    val dependencies: List<TransactionReferenceModel>
): InitialTransactionRequestModel()